msteams-platform/messaging-extensions/build-api-based-message-extension.md

```diff
@@ -0,0 +1,462 @@
+                                            "url": "${if(image, image, '')}",
+                                            "size": "Medium"
+                                        }
+                                    ]
+                                }
+                            ]
+                        },
+                        {
+                            "type": "FactSet",
+                            "facts": [
+                                {
+                                    "title": "Repair ID:",
+                                    "value": "${if(id, id, 'N/A')}"
+                                },
+                                {
+                                    "title": "Date:",
+                                    "value": "${if(date, date, 'N/A')}"
+                                }
+                            ]
+                        }
+                    ]
+                }
+            ]
+        },
+        "previewCardTemplate": {
+            "title": "Title: ${if(title, title, 'N/A')}",
+            "subtitle": "Description: ${if(description, description, 'N/A')}",
+            "text": "Assigned To: ${if(assignedTo, assignedTo, 'N/A')}",
+            "image": {
+                "url": "${if(image, image, '')}"
+            }
+        }
+    }
+    ```
+
+    </details>
+
+1. Update app manifest: Update the app manifest with the composeExtensions property. The following is an example of the app manifest with the `composeExtensions` property:
+
+   ```json
+      "composeExtensions": [
+        {
+          "type": "ApiBased",
+          "apiSpecFile": "listrepairsapispec.yaml",
+          "commands": [
+            {
+              "context": [
+                "compose"
+              ],
+              "type": "query",
+              "id": "listRepairs",
+              "title": "List repairs",
+              "parameters": [
+                {
+                  "title": "Filter",
+                  "name": "filter",
+                  "description": "Filter repairs by who they're assigned to."
+                }
+              ],
+              "apiResponseRenderingTemplateFile": "listrepairsresponsetemplate.json"
+            }
+          ]
+        }
+      ],
+      "validDomains": [
+        "repairs-api-2023.azurewebsites.net"
+      ]
+    }
+
+   ```
+
+   Ensure that the app has the following information:
+
+    * App name
+    * App color icon
+    * App outline icon
+    * Terms of Service URL
+    * Privacy policy URL
+    * The authentication type in `composeExtension.auth`: no auth, aad, oauth, or service_http.
+
+   **Query Parameters**
+ 
+   |Name  |Description  |
+   |---------|---------|
+   |`composeExtension.type`     |   We take an OpenAPI spec and it can either be used by BizChat to figure out how/when to call the API, or used by dev toolkit to help generate manifest/Adaptive Card templates.  Update the value as `ApiBased`. |
```

Comment:
I believe *BizChat* is internal-only term. Also, is this first sentence necessary to customer scenario completion, or more of an implementation detail of how our infra/tools use this value? Maybe simplify to:

For API-based message extensions, specify the value as `ApiBased`.

Reply:
Removed Biz chat content.